spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -193,6 +197,61 @@ public void onMessage(ConsumerRecord<Integer, String> message, Acknowledgment ac
 	}
 
 	@Test
+	public void testSlowConsumerCommitsAreProcessed() throws Exception {
+		Map<String, Object> props = KafkaTestUtils.consumerProps("slow", "false", embeddedKafka);
+		DefaultKafkaConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<Integer, String>(props);
+		ContainerProperties containerProps = new ContainerProperties(topic5);
+		containerProps.setAckCount(1);
+		containerProps.setPauseAfter(100);
+		containerProps.setAckMode(AckMode.MANUAL);
+		KafkaMessageListenerContainer<Integer, String> container =
+				new KafkaMessageListenerContainer<>(cf, containerProps);
+		final CountDownLatch latch = new CountDownLatch(6);
+		containerProps.setMessageListener(new AcknowledgingMessageListener<Integer, String>() {
+
+			@Override
+			public void onMessage(ConsumerRecord<Integer, String> message, Acknowledgment ack) {
+				logger.info("slow: " + message);
+				try {
+					Thread.sleep(1000);
+				}
+				catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+				}
+				ack.acknowledge();
+				latch.countDown();
+			}
+
+		});
+		container.setBeanName("testSlow");
+		container.start();
+		ContainerTestUtils.waitForAssignment(container, embeddedKafka.getPartitionsPerTopic());
+		Consumer<?, ?> consumer = spyOnConsumer(container);
+
+
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		ProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<Integer, String>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(topic5);
+		template.sendDefault(0, "foo");
+		template.sendDefault(2, "bar");
+		template.sendDefault(0, "baz");
+		template.sendDefault(2, "qux");
+		template.flush();
+		Thread.sleep(300);
+		template.sendDefault(0, "fiz");
+		template.sendDefault(2, "buz");
+		template.flush();
+
+		// Verify that commitSync is called when paused
+		assertThat(latch.await(60, TimeUnit.SECONDS)).isTrue();
+		InOrder inOrder = new InOrderImpl(Collections.singletonList(consumer));
```

Comment:
M-m-m. `InOrder` JavaDocs:
```
 * Allows verification in order. E.g:
 * 
 * <pre class="code"><code class="java">
 * InOrder inOrder = inOrder(firstMock, secondMock);
 * 
 * inOrder.verify(firstMock).add("was called first");
 * inOrder.verify(secondMock).add("was called second");
```
Since we deal only with one mock, we don't need this protection. Simple verify would be enough.

We have to `stop()` `container` in the end. 

Will fix everything on merge.